ext/afform/core/Civi/Api4/Action/Afform/SubmitFile.php

```diff
@@ -98,15 +98,20 @@ protected function processForm() {
 
     $file = civicrm_api3('Attachment', 'create', $attachmentParams);
 
-    // Update multi-record custom field with value
+    $updateParams = [
+      'values' => [$this->fieldName => $file['id']],
+    ];
+
     if (strpos($apiEntity, 'Custom_') === 0) {
-      civicrm_api4($apiEntity, 'update', [
-        'values' => [
-          $idField => $entityId,
-          $this->fieldName => $file['id'],
-        ],
-      ]);
+      // Update multi-record custom field with value
+      $updateParams['values'][$idField] = $entityId;
     }
+    else {
+      // Update entity field
+      $updateParams['where'] = [[$idField, '=', $entityId]];
```

Comment:
These are just two various ways of saying the same thing. They are not functionally different. During an update action, placing the `$id` in the `values` has the same effect as placing it in the `where`.

Reply:
Good to know. Changed.